src/components/streamSelect/StreamSelect.js

```diff
@@ -5,59 +5,81 @@ class StreamSelect extends Component {
+      { id: "5ddb98f5e4b0937e6a4507f2", name: "Livery Demo" },
+      { id: "5ddb986ee4b0937e6a4507e9-dev", name: "Robolingo" },
+    ];
+    let path = window.location.pathname;
+    let urlStreamID = path.replace("/", "");
+    let id;
+    if (path === "/") {
+      id = "5ddb98f5e4b0937e6a4507f2";
+    } else {
+      if (this.indexOfStreams(urlStreamID) > 0) {
+        id = urlStreamID;
+      } else {
+        id = "custom";
+      }
+    }
     this.state = {
-      currentStream: "5ddb98f5e4b0937e6a4507f2",
+      currentStream: id,
       streamIsCustom: false,
     };
   }
+  indexOfStreams(id) {
+    for (let i = 0; i < this.defaultStreams.length; i++) {
+      if (this.defaultStreams[i].id === id) {
+        return i;
+      }
+    }
+    return -1;
+  }
   updateField(event) {
     this.customStreamID = event.target.value;
   }
   setCustomStream() {
     this.props.setStream(this.customStreamID);
     this.setState({ currentStream: "custom", streamIsCustom: true });
+    window.history.pushState("LiveryDemo", "Livery Demo", this.customStreamID);
   }
   selectStream(event) {
     this.props.setStream(event.target.value);
     this.setState({ currentStream: event.target.value, streamIsCustom: false });
+    window.history.pushState("LiveryDemo", "Livery Demo", event.target.value);
   }
   render() {
     return (
       <div className="options-bar">
         <div className="stream-select-wrap">
           <div className="stream-select-dropdown">
-            <span>Select Stream: </span>
+            <span>Stream: </span>
             <select
               onChange={(e) => {
                 this.selectStream(e);
               }}
               value={this.state.currentStream}
             >
               <optgroup label="ExMG">
-                <option value="5c8b790e8f08e4ad1d1dc339-staging">
-                  Angry Bytes
-                </option>
-                <option value="5c52edb53e930320967a5d55-dev">Ex Machina</option>
-                <option value="5ddb98f5e4b0937e6a4507f2">Livery Demo</option>
-                <option value="5ddb986ee4b0937e6a4507e9-dev">Robolingo</option>
-                <option
-                  value="custom"
-                  // hidden={!this.state.streamIsCustom}
-                  disabled="disabled"
-                >
+                {this.defaultStreams.map((el) => (
+                  <option key={el.id} value={el.id}> {el.name} </option>
+                ))}
+                <option value="custom" disabled="disabled">
                   Custom
                 </option>
               </optgroup>
             </select>
           </div>
 
           <div className="stream-select-custom">
-            <span>Or ID: </span>
+            <span>ID: </span>
             <input
               onChange={(e) => {
                 this.updateField(e);
               }}
               placeholder="Stream ID"
+              maxlength="1000"
```

Comment:
Nitpicking, but, I'm curious, why this value? I guess it's ok if you need it for browser layout purposes (perhaps just replace it by a more modest/fitting number like 35). But otherwise I'd just leave it out rather than specifying an arbitrary large number.

Reply:
When putting a large value (over 2000) the site breaks because it's over the character limit of a URL. So because I didn't know the max length of a stream ID I just figured I'd put 1000 so it wouldn't be the cause of any errors.

Reply:
You could ask Leon or Joep to specify how long this value can get if you want to know and then add 8 for the longest possible suffix (-staging) to make this fit exactly. But for simplicity's sake, unless you need it for an immediate reason, I'd just try not specifying this property. I do vaguely recall this value also affecting how it's shown though, so you might indeed need to specify this. Anyhow.. Not really important; like I said; nit picking, sorry.